<?xml version="1.0"?>

<launch>
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find quadrotor_tunnel_nav)/config/rbpf/rosconsole.config"/>

  <!--SLAM RUN-->
  <node pkg="rviz" type="rviz" name="rviz_nav" output="screen" args="-d $(find quadrotor_tunnel_nav)/config/rbpf/rbpf.rviz"></node>

  <node pkg="quadrotor_tunnel_nav" type="mrpt_rbpf_slam" name="slam" output="screen">
    <param name="ini_filename" value="$(find quadrotor_tunnel_nav)/config/rbpf/grid_slam_demo.ini"/>
    <param name="gtruthw_frame_id" value="ground_truth/world"/>
    <param name="gtruthb_frame_id" value="ground_truth/base_link"/>
    <param name="odom_frame_id" value="nav"/>
    <param name="global_frame_id" value="map"/>
    <param name="base_frame_id" value="base_link"/>
    <!--for multiple sensors, place comma eg "r1/front_laser/scan,/beacon" -->
    <param name="sensor_source"
      value="/iris/depth/points,
             /iris/range_front,
             /iris/range_rfront,
             /iris/range_lfront,
             /iris/range_right,
             /iris/range_left,
             /iris/range_rrear,
             /iris/range_lrear,
             /iris/range_rear,
             /iris/range_up,
             /iris/range_ufront,
             /iris/range_urear,
             /iris/range_down,
             /iris/range_dfront,
             /iris/range_drear"/>
    <rosparam command="load" file="$(find quadrotor_tunnel_nav)/config/rbpf/rbpf_default.yaml" />
  </node>
</launch>
